### ISSUE TITLE
Prevent coursera-dl from downloading "extended-description-mixed" video files

### ISSUE THREAD

**Issue by raffam** [coursera-dl/coursera-dl#731]
🚨Please review the [Troubleshooting](../#troubleshooting) section
before reporting any issue. Don't forget to check also the current issues to
avoid duplicates.

### Subject of the issue
I want to download only actual lectures, withouth the "extended-description-mixed" lectures

### Expected behaviour
I want to download only actual lectures. To do so, i put a `-f "mp4"` commnad to download only video files. It actually works, but the problem is that, for every single lecture, it still downloads both the lecture and a video files with the suffix "extended-description-mixed", which is basically the lecture but with an additional voice telling you when the slide changes.
So for example, if the lecture is named "gradient descent", it downloads both "gradient_descent.mp4", which is what I want, and "gradient_descent- extended-description-mixed.mp4", which I don't want.

### Your environment
* Operating System (name/version): Windows 10
* Python version: 3.7
* coursera-dl version: commit 9b434bcf3c4011bf3181429fe674633ae5fb7d4d

### Steps to reproduce
Tell us how to reproduce this issue, please don't forget to include
enough information so that you can help us help you:

* Is the problem happening with the latest version of the script? Yes
* Do you have all the recommended versions of the modules? See them in the
  file `requirements.txt`. Yes
* What is the course that you are trying to access? `deep-learning` specialization
* What is the precise command line that you are using (don't forget to obfuscate
  your username and password, but leave all other information untouched).
`    coursera-dl -u ****** -p ****** --resume --specialization deep-learning --path=./coursera_scraped_hires --video-resolution 720p -f "mp4"`
* What are the precise messages that you get? Please, use the `--debug`
  option before posting the messages as a bug report. Please, copy and paste
  them.  Don't reword/paraphrase the messages.

I don't think that's relevant

### Actual behaviour
It downloads all the .mp4 files, including the ones with a suffix "extended-description-mixed" that I don't want

**Comment by raffam** [comment]
It looks like putting a `--lecture_filter "^((?!extended-description-mixed).)*$"` do the job, although the source code specifies that this option is going to be deprecated, don't know why

Let's see

**Comment by raffam** [comment]
Nada, it doesn't work. It still downloads the "extended-description-mixed" videos still

**Comment by deepghodasara1999** [comment]
please anyone can provide solution, i am also stuck here

**Comment by deepghodasara1999** [comment]
> It looks like putting a `--lecture_filter "^((?!extended-description-mixed).)*$"` do the job, although the source code specifies that this option is going to be deprecated, don't know why
> 
> Let's see

yes this will not work because in lecture name it is taking real name of lecture not filename to be download, so there is no meaning of using `lecture_filter`

**Comment by deepghodasara1999** [comment]
> It looks like putting a `--lecture_filter "^((?!extended-description-mixed).)*$"` do the job, although the source code specifies that this option is going to be deprecated, don't know why
> 
> Let's see

Can you explain that what this line `--lecture_filter "^((?!extended-description-mixed).)*$"` represent and what is do?

**Comment by dakdouky** [comment]
For those who are still looking around for a fix for this, see https://github.com/coursera-dl/coursera-dl/pull/791
